### ISSUE TITLE
Support for Quarkus and native image

### ISSUE THREAD

**Issue by adpande** [jdbi/jdbi#1890]
It would be awesome to see JDBI compatible with Quarkus and support native image. 

Ref: https://quarkus.io/guides/building-my-first-extension
https://quarkus.io/guides/writing-native-applications-tips

Thanks!

**Comment by hgschmie** [comment]
There are a number of projects that use quarkus with jdbi. If there are ways to build in specific (quarkus-native) code, we are glad to take a PR and add it to the jdbi code project. 

Some projects (like dropwizard) chose to maintain the jdbi integration as part of the project; I am not sure of the policies of the quarkus project

**Comment by stevenschlansker** [comment]
Native image support is tracked in #1797

**Comment by smil2k** [comment]
Hi,

I started an extension under the quarkiverse umbrella to add native image support using quarkus:

https://github.com/quarkiverse/quarkus-jdbi

**Comment by cies** [comment]
>  I could imagine a huge refactoring which is similar to the path drools took to support native images: they ditched reflection almost entirely, but I don't see the benefits overweight the effort required.

I dont think that refactor is that hard. Most of the reflection in JDBI is around bean/pojo mapping/binding. That should ideally split out into a package of it's own. So jdbi core becomes "bring your own mapper" with lots of optional addition packages for all kinds of mappers. It would reduce the size of `core` 3-6kLOC (quick estimation that 20-30% of the size of `core`)

**Comment by stevenschlansker** [comment]
Hi @adpande @smil2k and @cies , Jdbi supports `native-image` now out of the box, and provides metadata configuration for core required reflection operations. Thus we consider the native-image support complete for now.

There is a sample project with Quarkus available (from another thread): https://github.com/Eng-Fouad/jdbi-quarkus-native

Therefore, we think this issue can be closed as complete. If there is anything else specifically missing for native image or Quarkus support, please feel free to reopen this issue with a detailed description of remaining problems.

**Comment by smil2k** [comment]
@stevenschlansker One thing regarding the https://github.com/quarkiverse/quarkus-jdbi plugin is based on Eng-Fouad's work, extending it with some extra dynamic lookups.
